### PR TITLE
KK-403 | Add project, update models, add tests

### PR DESCRIPTION
**Pull request by quyenlq**
A pretty big PR since it's not easy to separate the changes without breaking the tests
So what have been changed from this PR are:
 - Added Project model, DB migration, schema
 - Added data migration to create the first project and add old children into the current project
 - Added project FK to child, event, venue
 - Added M2M relationship to users (for admin UI)
 - Update child`availableEvents/pastEvents` query to only return events belong to the child project
 - Update child mutation to auto add children to current project
 - Only send notifications to guardians of the same project when its new event published
 - Fixed unittest with `Project` thing

### CODE REVIEW

File: projects/migrations/0001_initial.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+from django.db import migrations, models
+
+
+def create_first_project(apps, schema_editor):
+    p = apps.get_model("projects", "Project").objects.create(year=2020)
+    p.save()
+
+
+def undo_create_first_project(apps, schema_editor):
+    apps.get_model("projects", "Project").objects.get(pk=1).delete()
```
[the remark this answers is not in the document]
Comment (quyenlq) [reply projects/migrations/0001_initial.py:13]: I believe you mean remove the whole `Project`. Undoing a migration file consists of different operations, each operation will be applied one by one in reverse order. In this case django will retry to reverse the data migration before dropping the Project table. If we don't have the reverse action for the data-migration operation, reverse migration is just doing nothing. So there will be a row in the DB and the drop table won't work

File: projects/tests/test_models.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+User = get_user_model()
+
+
+@pytest.mark.django_db
+def test_enrolment_creation(occurrence, project):
+    child = ChildFactory(project=project)
+    occurrence.children.add(child)
+    assert occurrence.children.count() == 1
+    assert child.occurrences.count() == 1
+    assert Enrolment.objects.count() == 1
```
[the remark this answers is not in the document]
Comment (quyenlq) [reply projects/tests/test_models.py:19]: Oops, copy-paste busted :D

File: projects/models.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    def __str__(self):
+        return f"#{self.pk} {self.year}"
+
+    @staticmethod
+    def get_or_create_current_project():
+        try:
+            return Project.objects.get(year=timezone.now().year)
+        except Project.DoesNotExist:
+            p = Project.objects.create(year=timezone.now().year)
+            return p
```
[the remark this answers is not in the document]
Comment (quyenlq) [reply projects/models.py:25]: Oh yeah, forgot to update this part -_-

File: projects/models.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+class Project(models.Model):
+    year = models.PositiveSmallIntegerField(verbose_name=_("year"), unique=True)
+    users = models.ManyToManyField(
+        "users.User", verbose_name=_("users"), related_name="projects", blank=True
+    )
+
+    class Meta:
+        verbose_name = _("project")
+        verbose_name_plural = _("projects")
+
```
Comment (tuomas777) [inline projects/models.py:15]: Maybe `ordering = ("year",)` ?

File: projects/models.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    def __str__(self):
+        return f"#{self.pk} {self.year}"
+
+    @classmethod
+    def get_or_create_current_project(cls):
+        return cls.objects.get_or_create(year=timezone.now().year)
```
Comment (tuomas777) [inline projects/models.py:21]: This doesn't seem to be needed anymore.

**Pull request merged**